### PR TITLE
feat: price range filter

### PR DESCRIPTION
**Pull request by harshalini**
Added price range filter
Clicking on categories on home page directs to products listing page with selected category
preview -
https://624c6c11a766c477a3f972f4--magenta-tapioca-c079d1.netlify.app/

### CODE REVIEW

File: src/components/filter/filter-comp/categoryFilter-comp/category-filter.js [hunk 1]
```diff
@@ -22,7 +26,11 @@ export function CategoryFilter() {
                     <label htmlFor="jeans-filter">Jeans</label>
                 </div>
             </div>
-            <span className="sort-heading">Women</span>
+            <input type = "checkbox" id="women-category"
+            onChange={(e) => e.target.checked ? dispatch({ type: "FOR", payload: "women" }) : dispatch({ type: "REMOVE-WOMEN" })}
```
Comment (JantuDeb) [inline src/components/filter/filter-comp/categoryFilter-comp/category-filter.js]: You can create  a `function`  here

File: src/components/filter/filter-comp/categoryFilter-comp/category-filter.js [hunk 2]
```diff
@@ -46,6 +54,11 @@ export function CategoryFilter() {
                     <label htmlFor="kurta-filter">kurta</label>
                 </div>
             </div>
+            <input type = "checkbox" id="men-category"
+           onChange={(e) => e.target.checked ? dispatch({ type: "FOR", payload: "shoes" }) : dispatch({ type: "REMOVE-SHOES" })}
```
Comment (JantuDeb) [inline src/components/filter/filter-comp/categoryFilter-comp/category-filter.js]: same here

**Review by JantuDeb — APPROVED** [review]
Overall the code structure and naming is good

File: src/components/categories/categories.js [hunk 3]
```diff
@@ -1,13 +1,26 @@
 import { useCategory } from "../../context/allContext";
 import { Link } from "react-router-dom";
+import { useSortedData } from "../../context/filter-context";
 export function GetCategories() {
   const { category } = useCategory();
+  const { state: {gCategories}, dispatch} = useSortedData();
   return (
     <div className="shop-by-categories">
```
Comment (mariyasada) [inline src/components/categories/categories.js]: why are you giving name as gCategories?

**Review by mariyasada — APPROVED** [review]
liked that you have created a separate file for routes. liked the code structure. one suggestion gives a proper name for categories if you want to.

File: src/components/categories/categories.js [hunk 4]
```diff
@@ -1,13 +1,26 @@
 import { useCategory } from "../../context/allContext";
 import { Link } from "react-router-dom";
+import { useSortedData } from "../../context/filter-context";
 export function GetCategories() {
   const { category } = useCategory();
+  const { state: {gCategories}, dispatch} = useSortedData();
```
Comment (Radheshyam1234) [inline src/components/categories/categories.js]: Naming of gCategories could be better here

File: src/components/filter/filter-comp/ratings-component/ratings-filter.js [hunk 5]
```diff
@@ -1,16 +1,16 @@
 import "./ratings.css"
 import { useSortedData } from "../../../../context/allContext"
 export function RatingsFilter() {
-    const { Pratings, dispatch } = useSortedData()
+    const { state: {Pratings}, dispatch } = useSortedData()
     return (
-        <div className="sort-by-rating">
+        <div className="sort-by-rating" style={{marginTop: "1rem"}}>
```
Comment (Radheshyam1234) [inline src/components/filter/filter-comp/ratings-component/ratings-filter.js:6]: Avoid inline styling

File: src/components/header/header.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ export function Header() {
             </p>
   
             <button className="btn v-btn explore-btn">
-              <Link to="/products">Explore now</Link>
+              <Link to="/products" style={{textDecoration: "none", color: "black"}}>Explore now</Link>
```
Comment (Radheshyam1234) [inline src/components/header/header.js]: Also here avoid using inline css

**Review by Radheshyam1234 — COMMENTED** [review]
Overall code looks good , Good work implementing reducer hook with dispatch

**Pull request merged**